packages/editor/src/components/rich-text/index.js

```diff
@@ -424,7 +424,13 @@ export class RichText extends Component {
 			return;
 		}
 
-		const { start, end } = this.createRecord();
+		const { start, end, formats } = this.createRecord();
+
+		if ( formats[ start ] ) {
```

Comment:
Could this be moved in `start !== this.state.start || end !== this.state.end` you think? Could we check right here which flag is set to avoid calling the prop?